bot-lib/src/commands/delete_class_category.rs

```diff
@@ -0,0 +1,41 @@
+use crate::commands::{get_channels, get_role};
+use crate::data::PoiseContext;
+use color_eyre::eyre::{OptionExt, Result};
+use regex::Regex;
+
+#[poise::command(
+    slash_command,
+    required_permissions = "MANAGE_CHANNELS",
+    description_localized("en-US", "Deletes a class category")
+)]
+pub async fn delete_class_category(
+    ctx: PoiseContext<'_>,
+    #[description = "The class number, eg. for CS2420 put in \"2420\""] number: u32,
+) -> Result<()> {
+    let guild = ctx.guild().ok_or_eyre("Couldn't get guild")?.id;
+    let channels = guild.channels(ctx).await?;
+
+    let category_and_role_name = format!("CS {}", number);
+    let category_channel = &get_channels(
+        ctx,
+        guild,
+        Regex::new(&format!("^{}$", category_and_role_name))?,
```

Comment:
Don't make a second string, just put this regex up in the top thing if you really need it